src/UWP/ArcGISRuntime.UWP.Samples/Samples/Analysis/ViewshedLocation/ViewshedLocation.xaml.cs

```diff
@@ -0,0 +1,129 @@
+
+            // Add the surface elevation.
+            Surface mySurface = new Surface();
+            mySurface.ElevationSources.Add(new ArcGISTiledElevationSource(_localElevationImageService));
+            myScene.BaseSurface = mySurface;
+
+            // Add the scene layer.
+            ArcGISSceneLayer sceneLayer = new ArcGISSceneLayer(_buildingsUrl);
+            myScene.OperationalLayers.Add(sceneLayer);
+
+            // Create the MapPoint representing the initial location.
+            MapPoint initialLocation = new MapPoint(-4.5, 48.4, 100.0);
+
+            // Create the location viewshed analysis.
+            _viewshed = new LocationViewshed(
+                initialLocation,
+                HeadingSlider.Value,
+                PitchSlider.Value,
+                HorizontalAngleSlider.Value,
+                VerticalAngleSlider.Value,
+                MinimumDistanceSlider.Value,
+                MaximumDistanceSlider.Value);
+
+            // Create a camera based on the initial location.
+            Camera camera = new Camera(initialLocation, 200.0, 20.0, 70.0, 0.0);
+
+            // Apply the camera to the scene view.
+            MySceneView.SetViewpointCamera(camera);
+
+            // Create an analysis overlay for showing the viewshed analysis.
+            _analysisOverlay = new AnalysisOverlay();
+
+            // Add the viewshed analysis to the overlay.
+            _analysisOverlay.Analyses.Add(_viewshed);
+
+            // Add the analysis overlay to the SceneView.
+            MySceneView.AnalysisOverlays.Add(_analysisOverlay);
+
+            // Update the frustum outline color.
+            // The frustum outline shows the volume in which the viewshed analysis is performed.
+            Viewshed.FrustumOutlineColor = Colors.Blue;
+
+            // Subscribe to tap events to enable moving the observer.
+            MySceneView.GeoViewTapped += MySceneViewOnGeoViewTapped;
+        }
+
+        private void MySceneViewOnGeoViewTapped(object sender, GeoViewInputEventArgs geoViewInputEventArgs)
+        {
+            // Update the viewshed location.
+            _viewshed.Location = geoViewInputEventArgs.Location;
+        }
+
+        private void HandleSettingsChange(object sender, RoutedEventArgs e)
+        {
+            // Return if viewshed hasn't been created yet. This happens when the sample is starting.
+            if (_viewshed == null)
+            {
+                return;
+            }
+
+            // Update the viewshed settings.
+            _viewshed.Heading = HeadingSlider.Value;
+            _viewshed.Pitch = PitchSlider.Value;
+            _viewshed.HorizontalAngle = HorizontalAngleSlider.Value;
+            _viewshed.VerticalAngle = VerticalAngleSlider.Value;
+            _viewshed.MinDistance = MinimumDistanceSlider.Value;
+            _viewshed.MaxDistance = MaximumDistanceSlider.Value;
+
+            // Return if the checkboxes are in an indeterminate.
```

Comment:
" ... are in an indeterminate **state**."
Is this necessary (can it happen?). What about just `_viewshed.IsVisible = AnalysisVisibilityCheck.IsChecked == True;`?

Reply:
The checkboxes use null to represent a partially checked state (checkbox is filled in rather than checked). This caught me by surprise, too. 

In the case of these samples, I don't think the check is actually needed. I get a warning from tooling about a possible `InvalidOperationException` if I remove it, but most users won't see that.